app/publish/tus.go

```diff
@@ -0,0 +1,222 @@
+package publish
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	"github.com/lbryio/lbrytv/app/auth"
+	"github.com/lbryio/lbrytv/app/proxy"
+	"github.com/lbryio/lbrytv/app/query/cache"
+	"github.com/lbryio/lbrytv/app/rpcerrors"
+	"github.com/lbryio/lbrytv/app/sdkrouter"
+	"github.com/lbryio/lbrytv/internal/errors"
+	"github.com/lbryio/lbrytv/internal/metrics"
+	"github.com/lbryio/lbrytv/internal/monitor"
+	"github.com/lbryio/lbrytv/internal/responses"
+	"github.com/ybbus/jsonrpc"
+
+	"github.com/sirupsen/logrus"
+	"github.com/tus/tusd/pkg/filestore"
+	tusd "github.com/tus/tusd/pkg/handler"
+)
+
+const module = "tus.publish"
+
+// PublishHandler handle media publishing on odysee-api, it implements TUS
+// specifications to support resumable file upload and extends the handler to
+// support fetching media from remote url.
+type PublishHandler struct {
+	*tusd.UnroutedHandler
+
+	logger       monitor.ModuleLogger
+	composer     *tusd.StoreComposer
+	authProvider auth.Provider
+}
+
+// NewHandler creates a new publish handler.
+func NewHandler(authProvider auth.Provider, uploadPath string) (*PublishHandler, error) {
+	h := &PublishHandler{}
+
+	if authProvider == nil {
+		return nil, fmt.Errorf("auth provider cannot be nil")
+	}
+
+	// make sure target directory is exist
+	defaultUploadPath := "./uploads"
+	if uploadPath == "" {
+		uploadPath = defaultUploadPath
+	}
+	if _, err := os.Stat(uploadPath); err != nil {
+		if os.IsNotExist(err) {
+			if err := os.MkdirAll(uploadPath, 0755); err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	// initialize tusd handler
+	composer := tusd.NewStoreComposer()
+	store := filestore.FileStore{
+		Path: uploadPath,
+	}
+	store.UseIn(composer)
+	h.composer = composer
+
+	config := tusd.Config{
```

Comment:
Feels wrong to define this configuration in the middle of the function, perhaps there's a better place for it?

Reply:
ok, i can pass this as a param and move it to `InstallRouter` since we need the base path there too